### PR TITLE
Runtime safety and style tweaks

### PR DESCRIPTION
**Pull request by rphmeier**

### CODE REVIEW

File: wasm-runtime/polkadot/src/codec/slicable.rs [hunk 1]
```diff
@@ -43,12 +44,14 @@ pub trait Slicable: Sized {
...
 
-impl<T: EndianSensitive> Slicable for T {
-	fn set_as_slice<F: FnOnce(&mut[u8]) -> bool>(fill_slice: F) -> Option<Self> {
-		let size = size_of::<T>();
-		let mut result: T = unsafe { uninitialized() };
+// note: the copy bound and static lifetimes are necessary for safety of `set_as_slice`.
+impl<T: Copy + EndianSensitive + 'static> Slicable for T {
+	fn set_as_slice<F: FnOnce(&mut [u8]) -> bool>(fill_slice: F) -> Option<Self> {
+		let size = mem::size_of::<T>();
+		let mut result: T = unsafe { mem::zeroed() };
```
Comment (gavofyork) [inline wasm-runtime/polkadot/src/codec/slicable.rs]: why is this better? it's obvious that it'll be written to; looks like it's just an invitation for the compiler's optimiser to fail.

File: wasm-runtime/polkadot/src/runtime/consensus.rs [hunk 2]
```diff
@@ -16,7 +16,7 @@
 
 //! Conensus module for runtime; manages the authority set ready for the native code.
 
-use runtime_support::Vec;
+use runtime_support::vec::Vec;
```
Comment (gavofyork) [inline wasm-runtime/polkadot/src/runtime/consensus.rs]: why not `use runtime_support::prelude::*;`?

File: wasm-runtime/polkadot/src/runtime/session.rs [hunk 3]
```diff
@@ -17,7 +17,7 @@
 //! Session manager: is told the validators and allows them to manage their session keys for the
 //! consensus module.
 
-use runtime_support::Vec;
+use runtime_support::vec::Vec;
```
Comment (gavofyork) [inline wasm-runtime/polkadot/src/runtime/session.rs]: same

File: wasm-runtime/polkadot/src/runtime/staking.rs [hunk 4]
```diff
@@ -16,7 +16,7 @@
 
 //! Staking manager: Handles balances and periodically determines the best set of validators.
 
-use runtime_support::Vec;
+use runtime_support::vec::Vec;
```
Comment (gavofyork) [inline wasm-runtime/polkadot/src/runtime/staking.rs]: same

File: wasm-runtime/polkadot/src/runtime/system.rs [hunk 5]
```diff
@@ -18,7 +18,8 @@
 //! and depositing logs.
 
 use primitives::{Block, BlockNumber, Hash, UncheckedTransaction, TxOrder, Hashable};
-use runtime_support::{Vec, swap};
+use runtime_support::mem;
+use runtime_support::vec::Vec;
```
Comment (gavofyork) [inline wasm-runtime/polkadot/src/runtime/system.rs]: same

File: wasm-runtime/polkadot/src/support/primitives.rs [hunk 6]
```diff
@@ -16,12 +16,12 @@
 
 //! Primitive types.
 
-use runtime_support::Vec;
+use runtime_support::vec::Vec;
```
Comment (gavofyork) [inline wasm-runtime/polkadot/src/support/primitives.rs]: same

File: wasm-runtime/polkadot/src/support/storable.rs [hunk 7]
```diff
@@ -36,7 +37,7 @@ pub trait Storable {
 /// Remove `key` from storage.
 pub fn kill(key: &[u8]) { runtime_support::set_storage(&twox_128(key)[..], b""); }
 
-impl<T: Default + Sized + EndianSensitive> Storable for T {
+impl<T: Default + Copy + EndianSensitive + 'static> Storable for T {
```
Comment (gavofyork) [inline wasm-runtime/polkadot/src/support/storable.rs]: why not add to `EndianSensitive` directly?

**Comment by gavofyork** [comment]
some comments.

File: wasm-runtime/polkadot/src/codec/slicable.rs (hunk 1 above)
Comment (rphmeier) [reply wasm-runtime/polkadot/src/codec/slicable.rs]: reading from zeroed memory is allowed. reading from uninitialized memory is undefined behavior. "reading" in this case includes writing without using `ptr::write`, which the `fill_slice` closure is probably expected to do.

Comment (gavofyork) [reply wasm-runtime/polkadot/src/codec/slicable.rs]: well, at least it'll likely be in L0 cache when the zeroes are being overwritten with the real data.

**Pull request merged**